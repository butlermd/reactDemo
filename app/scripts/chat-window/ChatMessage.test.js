'use strict';

jest.unmock('react-test-renderer');
jest.unmock('./ChatMessage');

import React from 'react';
import renderer from 'react-test-renderer';
import ChatMessage from './ChatMessage'

describe('ChatMessage', () => {
  var message, pending;

  beforeEach(() => {
    message = {
      text: 'chat text',
      user: 'username'
    };

    pending = false;
  });

  it('when the message is not pending', () => {
    const tree = renderer.create(<ChatMessage message={message} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('contains the text', () => {
    pending = true;

    const tree = renderer.create(<ChatMessage message={message} pending={pending} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});