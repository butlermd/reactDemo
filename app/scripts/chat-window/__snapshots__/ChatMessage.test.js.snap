exports[`ChatMessage contains the text 1`] = `
<div
  className="col-md-12"
  style={
    Object {
      "color": "#bbbbbb",
      "font-style": "italic",
    }
  }>
  <strong>
    username
    : 
  </strong>
   
  chat text
</div>
`;

exports[`ChatMessage when the message is not pending 1`] = `
<div
  className="col-md-12"
  style={Object {}}>
  <strong>
    username
    : 
  </strong>
   
  chat text
</div>
`;
